internal/core/config/monitor.go

```diff
@@ -73,11 +73,12 @@ type MonitorConfig struct {
 	OtherConfig map[string]interface{} `yaml:",inline" neverLog:"omit"`
 	// ValidationError is where a message concerning validation issues can go
 	// so that diagnostics can output it.
-	Hostname        string          `yaml:"-" json:"-"`
-	BundleDir       string          `yaml:"-" json:"-"`
-	ValidationError string          `yaml:"-" json:"-" hash:"ignore"`
-	MonitorID       types.MonitorID `yaml:"-" hash:"ignore"`
-	EnabledMetrics  []string        `yaml:"-" json:"-" hash:"ignore"`
+	Hostname          string          `yaml:"-" json:"-"`
+	BundleDir         string          `yaml:"-" json:"-"`
+	ValidationError   string          `yaml:"-" json:"-" hash:"ignore"`
+	MonitorID         types.MonitorID `yaml:"-" hash:"ignore"`
+	EnabledMetrics    []string        `yaml:"-" json:"-" hash:"ignore"`
+	EnabledMetricsSet map[string]bool `yaml:"-" json:"-" hash:"ignore"`
```

Comment:
Can remove this if you don't end up using it.

Reply:
Yes, I'll remove it

Reply:
Please remove this for sure, optimizations shouldn't really exist on the config structs, do that somewhere else if it really matters.